pallets/clusters/src/lib.rs

```diff
@@ -202,7 +202,7 @@ pub mod pallet {
 			let current_block_number = <frame_system::Pallet<T>>::block_number();
 
 			let cluster_id = blake2_128(
-				&[current_block_number.encode(), name.to_vec().clone(), extrinsic_index.clone().encode(), who.clone().encode()].concat(),
+				&[current_block_number.encode(), Vec::from(name.clone()), extrinsic_index.clone().encode(), who.clone().encode()].concat(),
```

Comment:
This changes nothing.
please use https://doc.rust-lang.org/std/slice/trait.Concat.html on slices, use `name.as_slice()` and encode is not needed for things that are already bytes etc.
@aang114 can you have a look please, we need to be max efficient here

Reply:
I already tried to use .as_slice, but I did not use it because it causes

```rust
 &[current_block_number.encode(), name.as_slice(), extrinsic_index.clone().encode(), who.clone().encode()].concat(),
      |                                                  ^^^^^^^^^^^^^^^- help: try using a conversion method: `.to_vec()`
      |                                                  |
      |                                                  expected struct `frame_support::dispatch::Vec`, found `&[u8]`

```
I can use `Vec::from(name.clone().as_slice())` instead.

Reply:
Ok changes something, missed the Vec::from part, but still there are a lot of allocations around

Reply:
Those `encode` are all needed because none of the variables passed in blake2_128 (`BlockNumber`, `u32`, `AccountId`) are bytes, as far as I can see.
I removed one `.clone()` from a `log.trace` that was not needed, but all the other allocations (i.e. `clone()`) are used where a reference could not be passed.
I don't see how can I optimize more. Maybe @aang114 can spot something more?